libqtile/bar.py

```diff
@@ -383,6 +383,11 @@ def finalize(self):
 
     def kill_window(self):
         """Kill the window when the bar's screen is no longer being used."""
+        for name in list(self.qtile.widgets_map.keys()):
+            w = self.qtile.widgets_map[name]
+            if w in self.widgets:
+                w.finalize()
+                del self.qtile.widgets_map[name]
```

Comment:
```suggestion
        for w, name in self.qtile.widgets_map.copy().items():
            if w in self.widgets:
                w.finalize()
                del self.qtile.widgets_map[name]
```

Reply:
Think you've got `w` and `name` the wrong way round there but, otherwise, yes!